projects/portal/src/app/models/faucets/faucet.guard.spec.ts

```diff
@@ -0,0 +1,154 @@
+import { ConfigService } from './../config.service';
+import { FaucetGuard } from './faucet.guard';
+import { TestBed } from '@angular/core/testing';
+import { UrlTree } from '@angular/router';
+import { of } from 'rxjs';
+
+const setup = (props?: { mockConfigService?: any }) => {
+  // Mock Values
+  const mockConfig = {
+    extension: {
+      faucet: [
+        {
+          denom: 'test_denom1',
+          network: 'test_network',
+          faucetURL: 'www/test.url',
+          hasFaucet: true,
+          creditAmount: 1200,
+          maxCredit: 1500,
+        },
+        {
+          denom: 'test_denom2',
+          network: 'test_network',
+          faucetURL: 'www/test.url',
+          hasFaucet: true,
+          creditAmount: 800,
+          maxCredit: 1000,
+        },
+      ],
+    },
+  };
+  const routerMock: any = {
+    queryParams: {
+      denom: 'test_denom1',
+      amount: '100',
+    },
+  };
+  const routerStateMock: any = {
+    snapshot: {},
+    url: '/faucet?address= &denom= &amount= ',
+  };
+
+  // Mock Service
+  const mockConfigService = {
+    config$: of(mockConfig),
+  };
+
+  // Setup TestBed
+  TestBed.configureTestingModule({
+    providers: [
+      FaucetGuard,
+      { provide: ConfigService, useValue: { ...mockConfigService, ...props?.mockConfigService } },
+    ],
+  });
+  const service = TestBed.inject(FaucetGuard);
+
+  return {
+    service,
+    routerMock,
+    routerStateMock,
+  };
+};
+
+describe('FaucetGuard when ConfigService returns a valid value', () => {
+  it('should be created', () => {
+    const { service } = setup();
+    expect(service).toBeTruthy();
+  });
+
+  it('returns TRUE when the config faucets and the queryParams have same denom', (done) => {
+    const { service, routerMock, routerStateMock } = setup();
+    const res = service.canActivate(routerMock, routerStateMock);
+    if (typeof res == 'boolean' || res instanceof UrlTree || res instanceof Promise) return;
```

Comment:
It is better if the impossible return value is also described in Error.
```ts
    if (typeof res == 'boolean' || res instanceof UrlTree || res instanceof Promise) {
      throw new Error('should return an Observable');
    }
```

Reply:
It should certainly return an error.